src/runtime/Module.cpp

```diff
@@ -282,7 +282,7 @@ Instance* Module::instantiate(ExecutionState& state, const ExternVector& imports
                          &data);
             }
 
-            if (UNLIKELY(elem->tableIndex() >= numberOfTableTypes() || index >= instance->m_tables[elem->tableIndex()]->size() || index + elem->functionIndex().size() > instance->m_tables[elem->tableIndex()]->size())) {
+            if (UNLIKELY(elem->tableIndex() >= numberOfTableTypes() || index > instance->m_tables[elem->tableIndex()]->size() || index + elem->functionIndex().size() > instance->m_tables[elem->tableIndex()]->size())) {
```

Comment:
It seems that `index >= instance->m_tables[elem->tableIndex()]->size()` can be totally removed

Reply:
Maybe adding an `(uint64_t)` casting could be used to avoid overflow situations.

Btw, isn't the wabt validator should be able to catch these issues? We could check it.

Reply:
Have you checked this?
It seems that wabt validator cannot check this case in advance.. but what do you think about it?